src/platform_linux/shmem.c

```diff
@@ -859,52 +1323,59 @@ platform_shm_find_free(shmem_info *shm,
    shm_lock_mem_frags(shm);
 
    uint32 shm_num_frags_tracked = shm->shm_num_frags_tracked;
-   uint32 shm_num_frags_inuse   = shm->shm_num_frags_inuse;
+   uint32 shm_num_frags_inuse   = shm->usage.nfrags_inuse;
 
    for (int fctr = 0; fctr < ARRAY_SIZE(shm->shm_mem_frags); fctr++, frag++) {
-      if (!frag->shm_frag_addr || (frag->shm_frag_size < size)) {
+      if (!frag->frag_addr || (frag->frag_size < size)) {
          continue;
       }
 
       // Skip fragment if it's still in-use
-      if (frag->shm_frag_freed_by_pid == 0) {
-         platform_assert((frag->shm_frag_freed_by_tid == 0),
+      if (frag->frag_freed_by_pid == 0) {
+         platform_assert((frag->frag_freed_by_tid == 0),
                          "Invalid state found for fragment at index %d,"
                          "freed_by_pid=%d but freed_by_tid=%lu "
                          "(which should also be 0)\n",
                          fctr,
-                         frag->shm_frag_freed_by_pid,
-                         frag->shm_frag_freed_by_tid);
+                         frag->frag_freed_by_pid,
+                         frag->frag_freed_by_tid);
 
          local_in_use++;
          continue;
       }
 
       // Fragment is free, but is it big enough for current request?
-      if (frag->shm_frag_size < size) {
+      if (frag->frag_size < size) {
          continue;
       }
       found_tracked_frag = TRUE;
       found_at_fctr      = fctr;
 
       // Record the process/thread to which free fragment is being allocated
-      frag->shm_frag_allocated_to_pid = getpid();
-      frag->shm_frag_allocated_to_tid = platform_get_tid();
+      frag->frag_allocated_to_pid = getpid();
+      frag->frag_allocated_to_tid = platform_get_tid();
 
-      shm->shm_num_frags_inuse++;
-      if (shm->shm_num_frags_inuse > shm->shm_num_frags_inuse_HWM) {
-         shm->shm_num_frags_inuse_HWM = shm->shm_num_frags_inuse;
+      shm->usage.nfrags_inuse++;
+      if (shm->usage.nfrags_inuse > shm->usage.nfrags_inuse_HWM) {
+         shm->usage.nfrags_inuse_HWM = shm->usage.nfrags_inuse;
```

Comment:
This does a "first-fit" search. It will be better to enhance this to "best-fit " search.
Otherwise, fragments freed from large fingerprint arrays, I've seen, get re-cycled for requests much smaller.

Not a deal breaker ... leave it for later ...